cf-oscore/src/main/java/org/eclipse/californium/oscore/ObjectSecurityLayer.java

```diff
@@ -75,17 +75,18 @@ public static Request prepareSend(OSCoreCtxDB ctxDb, Request message) throws OSE
 	 * @param ctxDb the OSCore context DB
 	 * @param message the message
 	 * @param ctx the OSCore context
-	 * @param newPartialIV boolean to indicate whether to use a new partial IV or not
-	 * @param outerBlockwise boolean to indicate whether the block-wise options
-	 *            should be encrypted or not
+	 * @param newPartialIV whether to use a new partial IV or not
+	 * @param outerBlockwise whether the block-wise options should be encrypted
+	 * @param requestSequenceNr sequence number (Partial IV) from the request
+	 *            (if encrypting a response)
 	 * 
 	 * @return the encrypted message
 	 * 
 	 * @throws OSException error while encrypting response
 	 */
 	public static Response prepareSend(OSCoreCtxDB ctxDb, Response message, OSCoreCtx ctx, final boolean newPartialIV,
-			boolean outerBlockwise) throws OSException {
-		return ResponseEncryptor.encrypt(ctxDb, message, ctx, newPartialIV, outerBlockwise);
+			boolean outerBlockwise, Integer requestSequenceNr) throws OSException {
```

Comment:
Is it intended, that the `Integer requestSequenceNr ` may be `null`? 
If so, would it be possible to add that to the javadoc?
If not, why is `Integer` used instead of `int`?

Reply:
True, requestSequenceNr does only have to be an Integer in the method Encryptor.encryptAndEncode, but i needlessly did that for other methods too. This has now been fixed in commit b5bc349413ab6d5572d45fff1fcce3c121431a65